File: stack/stack.py
class Stack:
    def __init__(self) -> None:
        self.items = []

    ## push element to the top of the stack
    def push(self, item):
        self.items.append(item)
        return f"{item} added to stack"

    ## remove element from the top of the stack
    def pop(self):
        if not self.is_empty():
            self.items.pop()
            return f"Item removed from stack"
        else:
            return IndexError("Empty stack")       

    ## get the size of the stack
    def size(self):
        return f"the stack has {len(self.items)} elements"

    ## look at the last element in the stack
    def peek(self):
        if not self.is_empty():
            return self.items[-1]
        else:
            return IndexError("Empty stack")

    ## check if stack is empty
    def is_empty(self):
        return len(self.items) == 0
        
stack = Stack()

print(stack.is_empty())
print(stack.push(10))
print(stack.push(12))
print(stack.push(3))
print(stack.size())
print(stack.peek())
print(stack.pop())
print(stack.peek())
print(stack.size())

File: stack/stack.js
class Stack {
  constructor() {
    this.items = [];
  }

  // check if stack is empty
  isEmpty() {
    return this.items.length === 0;
  }

  // push a new element to the stack
  push(item) {
    this.items.push(item);
    return `${item} added to stack`;
  }

  // remove the last element placed in the stack
  pop(item) {
    if (!this.isEmpty()) {
      this.items.pop();
      return "Item removed from stack";
    } else {
      throw new Error("Empty stack");
    }
  }

  // look at the last element in the stack
  peek() {
    if (!this.isEmpty()) {
      return this.items[this.items.length - 1];
    } else {
      throw new Error("Empty stack");
    }
  }

  // get the size of the stack
  size() {
    return this.items.length;
  }
}

const stack = new Stack();
console.log(stack.push(1));
console.log(stack.push(2));
console.log(stack.push(3));

console.log(stack.pop());
console.log(stack.peek());
console.log(stack.size());
